### PR TITLE
feat: demo path to migrate marketing collection away from stitching using same name w/ ENV flag

### PR DESCRIPTION
**Pull request by mzikherman**
This is a toy example that demonstrates:

  - adding a new type called `MarketingCollection` , which should have the same fields as the stitched type (and is called the same name)
  - adding a root field `marketingCollection: MarketingCollection` to point to this new resolver
  - using an ENV var to switch between schema/resolvers
 
As this uses an ENV var, and the MP server on boot builds an in-memory schema - after the ENV var is toggled MP must be rebooted. Since that's anticipated for ENV vars (`hokusai production refresh`), but indirect and not intuitive for Unleash (update Unleash flag, wait 15-30 seconds for it to fan out to clients, and _then_ `hokusai production refresh`) I think we should use an ENV var over Unleash for this.

The complexity around `_schema.graphql` file generation (and locally, if someone making a PR to this repo has that flag enabled or not) winds up being sidestepped by making sure that the new unstitched types + schema match exactly (the goal for backwards compatibility).

Then, the issue becomes: how do we prefer/merge in one version over another based on the ENV var? There's no easy magic solution here - all the places a field is being added to the schema need to become conditional - it seems doable to me!

Flag on - 
Screen Shot 2024-07-16 at 4 57 40 PM

Flag off -
Screen Shot 2024-07-16 at 4 58 19 PM

This can actually be safely merged if you'd like to then iterate on top of this, in this style @egdbear .

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "ci": "yarn test",
     "dev": "DEBUG=info,warn,error babel-node --extensions '.ts,.js' --inspect index.js",
     "dump-schema": "babel-node --extensions '.ts,.js' ./scripts/dump-schema.ts",
-    "dump:local": "yarn dump-schema _schemaV2.graphql & wait",
+    "dump:local": "USE_UNSTITCHED_MARKETING_COLLECTION_SCHEMA=false yarn dump-schema _schemaV2.graphql & wait",
```
Comment (mzikherman) [inline package.json:20]: This is b/c we know that the `_schemaV2.graphql` file won't meaningfully change based on this flag being enabled or not. However - without this we might see noisy updates to the file toggle back and forth as people (who may have the flag enabled or not) make PR's to Metaphysics - the updates to the file are more whitespace/auto-generated based on which schema is getting used, and so we can/should suppress that for this purpose.

File: src/lib/stitching/gravity/schema.ts [hunk 2]
```diff
@@ -7,18 +7,22 @@ import {
...
   "artistSeries",
   "artistSeriesConnection",
   "curatedMarketingCollections",
   "marketingCategories",
-  "marketingCollection",
   "marketingCollections",
   "viewingRoom",
   "viewingRooms",
-]
+].concat(
```
Comment (mzikherman) [inline src/lib/stitching/gravity/schema.ts:21]: Eventually `marketingCollections` gets added here as well besides `marketingCollection`.

File: src/schema/v2/schema.ts [hunk 3]
```diff
@@ -235,6 +237,12 @@ const OptionalFieldDirective = new GraphQLDirective({
   locations: [DirectiveLocation.FIELD],
 })
 
+const marketingCollectionUnstitchedRootField = config.USE_UNSTITCHED_MARKETING_COLLECTION_SCHEMA
```
Comment (mzikherman) [inline src/schema/v2/schema.ts:240]: This approach will get used based on the flag inside of https://github.com/artsy/metaphysics/blob/6fb7293eff13b17e1472150be114b0c69c899cb3/src/lib/stitching/gravity/v2/stitching.ts#L77 as well (for example) - conditionally include that schema stitching snippet, or not.

File: package.json (hunk 1 above)
Comment (damassi) [reply package.json:20]: So with `USE_UNSTITCHED_MARKETING_COLLECTION_SCHEMA`, we can toggle this to on while developing the schema migration, and then once complete remove all references from the codebase and flatten things out in terms of logic?

Comment (mzikherman) [reply package.json:20]: Yep - at the end when we're all done having an unstitched schema that's based on the flag - I do anticipate a fair number of occurrences of the string `USE_UNSTITCHED_MARKETING_COLLECTION_SCHEMA` in the repo. Probably concentrated in several files, but definitely (like we should always be doing!) - we shouldn't enable the flag and walk away - but enable the flag, and within 1-2 sprints after that - come back here and do the cleanup once we know we're good.

I was seeing relatively meaningless updates to the `_schema` file locally based on the flag, so I'm just predicting confusion from wider devs ('wait why did this change, i dont have the flag enabled'), plus this CI script - https://github.com/artsy/metaphysics/blob/6fb7293eff13b17e1472150be114b0c69c899cb3/scripts/ensure-schema-update.sh#L14 - will also confusingly fail back and forth.

Comment (anandaroop) [reply package.json:20]: Nice, and it seems like we can check our work — i.e. verifying that the migration is fully schema-compatible with the stitched version — by dumping the schema with `true` and `false` values for the env var, and confirming that there is no diff in the resulting files.

(With the caveat that this wouldn't actually verify the behavior of any new resolvers, but presumably test coverage comparable to what's in Gravity now would do that.)

**Review by anandaroop — APPROVED** [review]
NIce! 

I was curious about what it would mean to avoid the V2 approach ([discussed](https://artsy.slack.com/archives/C2TQ4PT8R/p1721160555255219?thread_ts=1721160397.234129&cid=C2TQ4PT8R) in Slack). Overall this feels good, but I guess we lose the ability to check old and new schema behavior side by side in the same graphql query, which I expect would be a nice dev-time convenience. But if your local MP is pointing to staging env, then you can still more or less do this by using staging MP for comparison, so I think this seems pretty workable 👍🏽

**Review by egdbear — APPROVED** [review]

**Comment by damassi** [comment]
Nice 👍 👍

**Pull request merged**